tests/unit-tests/test_nn_model.py

```diff
@@ -932,11 +1042,105 @@ def test_hgt_with_edge_features(input_dim, output_dim, dev):
     init_hgtlayer(layerwithef)
     layerwithef = layerwithef.to(dev)
     layerwithef.eval()
-    dev_emb = layerwithef(block, node_feats, edge_feats)
+    div_emb = layerwithef(block, node_feats, edge_feats)
     assert np.not_equal(baseline_emb['n1'].detach().cpu().numpy(), 
-                        dev_emb['n1'].detach().cpu().numpy()).any()
+                        div_emb['n1'].detach().cpu().numpy()).any()
+
+
+    mp_ops = ['add', 'concat', 'sub', 'mul', 'div']
+    for e_f_mp_op in mp_ops:
+        # test case 2: normal case, one edge type has features
+        # r0 case
+        layerwithef = HGTLayerwithEdgeFeat(
+            input_dim, output_dim, 
+            ntypes, etypes,
+            num_heads=4,
+            edge_feat_name={("n0", "r0", "n1"): ['feat']},
+            edge_feat_mp_op=e_f_mp_op,
+            activation=th.nn.ReLU(),
+            dropout=0.0,
+            norm='')
+        init_hgtlayer(layerwithef)
+        layerwithef = layerwithef.to(dev)
+        layerwithef.eval()
+
+        edge_feats = {
+            ("n0", "r0", "n1"): th.rand(r0_eid.shape[0], input_dim).to(dev)
+        }
+
+        emb = layerwithef(block, node_feats, edge_feats)
+        
+        assert 'n0' not in emb
+        assert emb['n1'].shape[0] == len(seeds['n1'])
+        assert emb['n1'].shape[1] == output_dim 
+
+        # r1 case
+        layerwithef = HGTLayerwithEdgeFeat(
+            input_dim, output_dim, 
+            ntypes, etypes,
+            num_heads=4,
+            edge_feat_name={("n0", "r1", "n1"): ['feat']},
+            edge_feat_mp_op=e_f_mp_op,
+            activation=th.nn.ReLU(),
+            dropout=0.0,
+            norm='')
+        init_hgtlayer(layerwithef)
+        layerwithef = layerwithef.to(dev)
+        layerwithef.eval()
+
+        edge_feats = {
+            ("n0", "r1", "n1"): th.rand(r1_eid.shape[0], input_dim).to(dev)
+        }
+
+        emb = layerwithef(block, node_feats, edge_feats)
+        
+        assert 'n0' not in emb
+        assert emb['n1'].shape[0] == len(seeds['n1'])
+        assert emb['n1'].shape[1] == output_dim
+    
```

Comment:
Could you add aother normal cases, where initialize `HGTLayerwithEdgeFeat` with both two edge types, but only provide one type of edge feature in forward(), which should also work.